sbndcode/OpDetSim/TriggerEmulationService.cc

```
@@ -44,22 +46,37 @@ namespace calib {
           }
 
           std::map<int, const raw::OpDetWaveform*> channel_to_waveform;
-          for (const auto& wvf : fWaveforms)
+          for (const auto& wvf : fWaveforms) {
               channel_to_waveform[wvf.ChannelNumber()] = &wvf;
+          }
 
           std::vector<int> Pair2 = { 6,   8,  10,  12,  14,  16,  36,  38,  40,  60,  62,  66,  68, 70,  84,  86,  88,  90,  92,  94, 114, 116, 118, 138, 140, 144, 146, 148, 162, 164, 168, 170, 172, 192, 194, 196, 216, 218, 220, 222, 224, 226, 240, 242, 246, 248, 250, 270, 272, 274, 294, 296, 298, 300, 302, 304};
           std::vector<int> Pair1 = { 7,   9,  11,  13,  15,  17,  37,  39,  41,  61,  63,  67,  69, 71,  85,  87,  89,  91,  93,  95, 115, 117, 119, 139, 141, 145, 147, 149, 163, 165, 169, 171, 173, 193, 195, 197, 217, 219, 221, 223, 225, 227, 241, 243, 247, 249, 251, 271, 273, 275, 295, 297, 299, 301, 303, 305};
           std::vector<int> Unpaired = {65,  64, 143, 142, 167, 166, 245, 244};
           std::set<int> used_channels;
 
           // resize
-          int ReadoutSize = fWaveforms[0].size();
+          int ReadoutSize;
+          if (PMT_Channels.empty()) { std::cout<<"Warning: Please provide PMT channels list."<<std::endl; ReadoutSize = fWaveforms[Pair1[0]].size(); }
+          else ReadoutSize = fWaveforms[PMT_Channels[0]].size();
           MonPulse->assign(ReadoutSize, 0);
 
           for (size_t i = 0; i < Pair1.size(); ++i) {
               int ch1 = Pair1[i];
               int ch2 = Pair2[i];
 
+              // check that ch1 and ch2 show up in PMT_Channels
+              if (PMT_Channels.empty()) std::cout<<"Warning: Please provide PMT channels list to check if channels are PMT channels."<<std::endl;
+              else {
+                  if (!(std::find(PMT_Channels.begin(), PMT_Channels.end(), ch1) != PMT_Channels.end())) {
+                      std::cout<<"Paired channel "<<ch1<<" is not PMT channel. Check Pairs list. Skipping..."<<std::endl;
```

This one (and the ones following) should probably throw an exception if the PMT channels provided do not match the pairing scheme.